azurerm/internal/services/vmware/vmware_cluster_resource.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/vmware/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceVmwareCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVmwareClusterCreate,
+		Read:   resourceVmwareClusterRead,
+		Update: resourceVmwareClusterUpdate,
+		Delete: resourceVmwareClusterDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Hour),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Hour),
+			Delete: schema.DefaultTimeout(5 * time.Hour),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ClusterID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"private_cloud_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.PrivateCloudID,
+			},
+
+			"cluster_size": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(3, 16),
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"av20",
+					"av36",
+					"av36t",
+				}, false),
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"hosts": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+func resourceVmwareClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Vmware.ClusterClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	privateCloudId, err := parse.PrivateCloudID(d.Get("private_cloud_id").(string))
+	if err != nil {
+		return err
+	}
+
+	id := parse.NewClusterID(subscriptionId, privateCloudId.ResourceGroup, privateCloudId.Name, name).ID()
+
+	existing, err := client.Get(ctx, privateCloudId.ResourceGroup, privateCloudId.Name, name)
```

Comment:
wouldn't
```suggestion
	existing, err := client.Get(ctx, id.ResourceGroup, id.Name, name)
```

also work and be cleaner?